applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/instances.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+'require form';
+'require uci';
+'require fs';
+'require network';
+'require rpc';
+'require shadowsocks-libev as ss';
+
+var conf = 'shadowsocks-libev';
+var cfgtypes = ['ss_local', 'ss_redir', 'ss_server', 'ss_tunnel'];
+
+var callServiceList = rpc.declare({
+	object: 'service',
+	method: 'list',
+	params: [ 'name' ],
+	expect: { '': {} }
+});
+
+return L.view.extend({
+	render: function(stats) {
+		var m, s, o;
+
+		m = new form.Map(conf,
+			_('Local Instances'),
+			_('Instances of shadowsocks-libev components, e.g. ss-local, \
+			   ss-redir, ss-tunnel, ss-server, etc.  To enable an instance it \
+			   is required to enable both the instance itself and the remote \
+			   server it refers to.'));
+
+		s = m.section(form.GridSection);
+		s.addremove = true;
+		s.cfgsections = function() {
+			return this.map.data.sections(this.map.config)
+				.filter(function(s) { return cfgtypes.indexOf(s['.type']) !== -1; })
+				.map(function(s) { return s['.name']; });
+		};
+		s.sectiontitle = function(section_id) {
+			var s = uci.get(conf, section_id);
+			return (s ? s['.type'] + '.' : '') + section_id;
+		};
+		s._renderSectionAdd = s.renderSectionAdd;
+		s.renderSectionAdd = function(extra_class) {
+			var el = this._renderSectionAdd(extra_class),
```

Comment:
Please use `form.GridSection.prototype.renderSectionAdd.apply(this, arguments);` instead of caching `_renderSectionAdd`.